processor/src/main/java/org/mapstruct/ap/internal/model/BeanMappingMethod.java

```diff
@@ -1055,7 +1089,7 @@ private boolean handleDefinedNestedTargetMapping(Set<String> handledTargets, Typ
                 .existingVariableNames( existingVariableNames )
                 .build();
 
-            unprocessedSourceParameters.removeAll( holder.getProcessedSourceParameters() );
+            holder.getProcessedSourceParameters().forEach( unprocessedSourceParameters::remove );
```

Comment:
Why is this changed?

Reply:
IntelliJ shows a warning: `Call to 'set.removeAll(list)' may work slowly` therefore I applied the quick fix to prevent a possible performance problem. I wasn't aware of this until that moment. More information here: https://stackoverflow.com/q/28671903/1213544

I honestly don't know why I only fixed this warning, especially because that part of the code is unrelated to this ticket. Therefore I reverted it.

Reply:
Interesting. I didn't know about this. The problem there is the fact that the `holder` returns a `List<Parameter>` which doesn't make much sense to be honest. It can be a `Collection` (and then the problem goes away). The thing is anyways that this parameters are the source parameters of a mapping method, so I doubt that we will ever go in the range where this can become a problem. Seems like a new warning from IntelliJ though, I haven't seen it before.

We can fix it in a different PR. I prefer like that, because it makes it easier to review the code and to focus on the main functionality in a single PR